crates/rpc/src/method/get_events.rs

```diff
@@ -370,7 +369,15 @@ fn map_to_block_to_number(
 
             Ok(Some(number))
         }
-        Some(Number(number)) => Ok(Some(number)),
+        Some(Number(number)) => {
+            let number = tx
+                .block_id(number.into())
+                .context("Querying block number")?
+                .ok_or(GetEventsError::BlockNotFound)?
+                .0;
+
+            Ok(Some(number))
+        }
```

Comment:
Isn't this a change in behavior? Looks like you can't use not-yet-existing block numbers as the upper limit now, while it was possible to do that before.